pkg/deploy/service.go

```diff
@@ -265,7 +269,27 @@ func buildStrategy(configStrategies map[string]model.Strategy, strategyName stri
 	return nil, fmt.Errorf("%s is not a valid strategy; define canary or blue-green strategy", strategyName)
 }
 
-func createDeploymentCanaryAnalysisStep(analysis *model.AnalysisStep) (*de.AnalysisAnalysisStepInput, error) {
+func createDeploymentCanaryAnalysisStep(analysis *model.AnalysisStep, analysisConfig *model.AnalysisConfig) (*de.AnalysisAnalysisStepInput, error) {
+	if analysisConfig == nil {
+		return nil, errors.New("analysis step is present but a top-level analysis config is not defined")
+	}
+
+	if analysisConfig.Queries == nil {
+		return nil, errors.New("top-level analysis config is present but no queries are defined")
+	}
+
+	for _, query := range *analysis.Queries {
+		var exists bool
+		for _, configQuery := range *analysisConfig.Queries {
```

Comment:
It might not be worth it but you could refactor this into
```go
queryConfig := findByName(analysisConfig.Queries, query)
if queryConfig == nil {
  return nil, fmt.Errorf("...")
}
```